### PR TITLE
Add code quality checks

### PR DESCRIPTION
**Pull request by ylep**

### CODE REVIEW

File: capsul/engine/database_json.py [hunk 1]
```diff
@@ -70,7 +71,10 @@ def set_path_metadata(self, path, metadata, named_directory=None):
         path = metadata['path']
         named_directory = metadata['named_directory']
         
-        self.json_dict.setdefault('path_metadata', {})[(named_directory, path)] = doc
```
Comment (ylep) [inline capsul/engine/database_json.py]: @sapetnioc I tried to fix this incorrect code, but I am not sure I understand what it is supposed to do. Please check if my fix is correct.

**Comment by ylep** [comment]
Hi @denisri and @sapetnioc, I need your help to resolve remaining flake8 issues in capsul. Some of the code is broken, and I cannot decide what is the good way of fixing it, or if we should even drop it entirely. Can you please have a look at the remaining errors reported by `flake8`?

**Comment by ylep** [comment]
Hi @denisri and @sapetnioc, I need your help to resolve remaining flake8 issues in capsul. Some of the code is broken, and I cannot decide what is the good way of fixing it, or if we should even drop it entirely. Can you please have a look at the remaining errors reported by `flake8`?

**Comment by ylep** [comment]
Hi @denisri and @sapetnioc, I need your help to resolve remaining flake8 issues in capsul. Some of the code is broken, and I cannot decide what is the good way of fixing it, or if we should even drop it entirely. Can you please have a look at the remaining errors reported by `flake8`?

**Comment by ylep** [comment]
Hi @denisri and @sapetnioc, I need your help to resolve remaining flake8 issues in capsul. Some of the code is broken, and I cannot decide what is the good way of fixing it, or if we should even drop it entirely. Can you please have a look at the remaining errors reported by `flake8`?

The reported errors are:
```
capsul/in_context/fsl.py:72:15: F821 undefined name 'Popen'
capsul/sphinxext/resources/numpy_ext/docscrape_sphinx.py:9:8: F821 undefined name 'six'
capsul/engine/database_populse.py:90:83: F821 undefined name 'doc'
capsul/engine/database_populse.py:117:5: F811 redefinition of unused 'set_path_metadata' from line 85
capsul/engine/database_populse.py:149:5: F811 redefinition of unused 'path_metadata' from line 94
capsul/process/process.py:1182:36: F821 undefined name 'srcdir'
capsul/study_config/config_modules/nipype_config.py:23:1: F404 from __future__ imports must occur at the beginning of the file
capsul/pipeline/pipeline_nodes.py:622:21: F821 undefined name 'deet_plugs'
capsul/in_context/spm.py:57:15: F821 undefined name 'Popen'
capsul/engine/__init__.py:373:48: F821 undefined name 'name'
capsul/engine/__init__.py:376:48: F821 undefined name 'name'
capsul/pipeline/pipeline_tools.py:446:19: F821 undefined name 'nodes_pos'
capsul/pipeline/pipeline_tools.py:448:63: F821 undefined name 'scale'
capsul/pipeline/pipeline_tools.py:449:64: F821 undefined name 'scale'
capsul/pipeline/pipeline_tools.py:452:69: F821 undefined name 'scale'
capsul/pipeline/pipeline_tools.py:453:70: F821 undefined name 'scale'
capsul/pipeline/pipeline_tools.py:958:50: F821 undefined name 'src'
capsul/pipeline/pipeline_tools.py:961:43: F821 undefined name 'src'
capsul/subprocess/spm.py:76:18: F821 undefined name 'last_line'
capsul/subprocess/spm.py:77:21: F821 undefined name 'last_line'
capsul/subprocess/spm.py:77:32: F821 undefined name 'last_line'
capsul/subprocess/spm.py:232:37: F821 undefined name 'batch_file'
capsul/pipeline/python_export.py:325:21: F821 undefined name 'step_node'
```

File: capsul/attributes/test/test_attributed_process.py [hunk 2]
```diff
@@ -40,7 +41,7 @@ class DummyListProcess(Process):
 
     def _run_process(self):
         open(self.result, 'w').write(
-            '{\n    truc=%s,\n    bidule=%s\n}' % truc, bidule)
+            '{\n    truc=%s,\n    bidule=%s\n}' % 'truc', 'bidule')
```
Comment (denisri) [inline capsul/attributes/test/test_attributed_process.py]: should rather be:
```suggestion
            '{\n    truc=%s,\n    bidule=%s\n}' % (self.truc, self.bidule))
```

File: capsul/sphinxext/resources/numpy_ext/docscrape_sphinx.py [hunk 3]
```diff
@@ -1,14 +1,12 @@
...
 import sys, re, inspect, textwrap, pydoc
 import sphinx
 import collections
 from .docscrape import NumpyDocString, FunctionDoc, ClassDoc
 
-if sys.version_info[0] >= 3:
-    sixu = lambda s: s
-else:
-    sixu = lambda s: unicode(s, 'unicode_escape')
+sixu = six.u
```
Comment (denisri) [inline capsul/sphinxext/resources/numpy_ext/docscrape_sphinx.py]: needs
```suggestion
import six
```

File: capsul/study_config/config_modules/nipype_config.py [hunk 4]
```diff
@@ -15,6 +16,9 @@
 ---------------------
 '''
 
+import glob
+import os
+
 # TRAITS import
 from __future__ import absolute_import
```
Comment (denisri) [inline capsul/study_config/config_modules/nipype_config.py]: should be moved before other imports

**Comment by denisri** [comment]
I pushed some fixes on master branch, that you should merge. They should fix several of your flake8 fails. Some still remain because I don't know about them:
capsul/engine/database_populse.py: @sapetnioc  ?
Others are suggested in comments above.

File: capsul/engine/database_json.py (hunk 1 above)
Comment (sapetnioc) [reply capsul/engine/database_json.py]: That was ok. There was another mistake just above that I fixed.

**Comment by sapetnioc** [comment]
I think this PR should import master changes. I wonder if it may fix some testing errors. For instance there is still Python 2.6 used in tests in this PR. And there is an error on six version (ensure_str missing). Is there someone on this ? Otherwise I can handle it.

File: capsul/study_config/config_modules/nipype_config.py [hunk 5]
```diff
@@ -1,11 +1,4 @@
 # -*- coding: utf-8 -*-
-##########################################################################
```
Comment (ylep) [inline capsul/study_config/config_modules/nipype_config.py]: Hi @sapetnioc, is there a reason why you deleted these licence comments in some files?

Comment (sapetnioc) [reply capsul/study_config/config_modules/nipype_config.py]: Well, I'd like to remove them all but I do it only when I edit a file and see it. For me we should not use headers in all files with elements that may vary on time such as date, maintainer, institution name, etc. And for the license, I think a single file is sufficient and much more easy to manage.

Comment (ylep) [reply capsul/study_config/config_modules/nipype_config.py]: Okay, I thought this was a mistake or a side-effect of `modernize` so I restored quite a few of these headers in aims.

Maybe we should open a dedicated [issue on brainvisa.github.io](https://github.com/brainvisa/brainvisa.github.io/issues) and discuss that issue there?

**Comment by ylep** [comment]
We are getting close: the only remaining flake8 issues are listed below. @sapetnioc do you know how we should fix those?

```
capsul/engine/__init__.py:373:48: F821 undefined name 'name'
capsul/engine/__init__.py:376:48: F821 undefined name 'name'
capsul/engine/database_populse.py:90:83: F821 undefined name 'doc'
capsul/engine/database_populse.py:117:5: F811 redefinition of unused 'set_path_metadata' from line 85
capsul/engine/database_populse.py:149:5: F811 redefinition of unused 'path_metadata' from line 94
```

**Comment by sapetnioc** [comment]
Some files I fixed are a mess because they are in an early development stage. But I tried to make them clean at least.

**Pull request merged**